core/src/jsMain/kotlin/ScannerBuilder.kt

```diff
@@ -15,7 +36,7 @@ public actual class ScannerBuilder {
 
     internal actual fun build(): PlatformScanner = BluetoothWebBluetoothScanner(
         bluetooth = bluetooth,
-        filters = filters.orEmpty(),
+        filters = filters?.convertDeprecatedFilters() ?: filterPredicates,
```

Comment:
These are non-optional, so it would have to be `filterPredicates.takeUnless { it.isEmpty() } ?: filters?.convertDeprecatedFilters() ?: emptyList()`.

For this deprecation case I don't _really_ think it matters too much, but my brief thought process on this was that, in order to use the new API, one must first delete the old API and not allow it to linger around. So once a consumer has migrated, a future hard-deprecation release is guaranteed to be a no-op for this specific item.

But also happy to invert it if you prefer - is a very minor lift on the consumer either way.

Reply:
Oh, I see you're reasoning. Sounds good. Let's keep it how you have it. 👍